### DISCUSSION TITLE
Split v8go C++ and header file into multiple files?

### DISCUSSION THREAD

**Discussion by stroiman** [tommie/v8go#74] [General]
When extending functionality, I need to modify `v8go.h` and `v8go.cpp`. 

I wonder why it's one big file, rather than split up, e.g., an `isolate.cpp`/`isolate.h` for code that is used from `isolate.go`, etc.

For console interception, I need to have an `Inspector` and `InspectorClient`. I started placing the C++ code in it's own set of files, and that didn't seem to give any problems.

I think the code would be much easier to navigate. Particularly because I can easily create editor commands for navigating between same-named `.go`, `.cc`, and `.h` files.

But I have no knowledge of CGo except from what I learned working with this code. I don't know if there's a compelling reason for having it in one big file.

Opinions on splitting the v8go C++ and header file into multiple smaller files?

If positive, I think I'd start by making a PR where `Isolate` and `Context` are extracted. Those two are what I depend on for the inspector functionality.

**Comment by tommie** [comment]
I've had the same question. It probably just grew slowly, and the interested people were fine with it.

Splitting it sounds good to me. I don't remember if you have to have a Go file to go with every .cc file, or if it just grabs all of them, but I guess we'll find out. :)
